hop/subscribe.py

```diff
@@ -72,12 +113,20 @@ def _main(args=None):
     # set offset
     start_offset = "earliest" if args.earliest else "latest"
 
+    # set timeout
+    t = args.timeout
+    timeout = None if t == "-1" else float(t)
+
     # read from topic
 
     # assume json format for the gcn
     gcn_format = "json"
 
     stream = Stream(format=gcn_format, config=config, start_at=start_offset)
     with stream.open(args.url, "r") as s:
-        for gcn_dict in s(timeout=args.timeout):
-            print_gcn(gcn_dict, args.json)
+        for msg in s(timeout=timeout):
+            gcn = classify_msg(msg)
+            if gcn is not None:
+                print_gcn(gcn, args.json)
+            else:
+                continue
```

Comment:
I don't think you need this else statement.

Reply:
That's right, thanks. Removed.